setup.py

```diff
@@ -5,7 +5,24 @@
 import os
 from os.path import join as ospj
 import io
-import imp
+try:
+    # Already obsolete, deprecated, and undocumented by Python 3.11, this was
+    # removed in Python 3.12 along with the rest of the imp module.
```

Comment:
You won't want to believe it, but I have some dedicated applications with a built-in Py2.7 interpreter ... from which this module originally came :-) :-)